src/intervalmap.rs

```diff
@@ -0,0 +1,1524 @@
+        let x_right = self.node_ref(x, Node::right);
+        if self.max(x_left) > self.max(x) {
+            self.node_mut(
+                x,
+                Node::set_max_index(self.node_ref(x_left, Node::max_index)),
+            );
+        }
+        if self.max(x_right) > self.max(x) {
+            self.node_mut(
+                x,
+                Node::set_max_index(self.node_ref(x_right, Node::max_index)),
+            );
+        }
+    }
+
+    /// Find the node with the minimum interval.
+    fn tree_minimum(&self, mut x: NodeIndex<Ix>) -> NodeIndex<Ix> {
+        while !self.left_ref(x, Node::is_sentinel) {
+            x = self.node_ref(x, Node::left);
+        }
+        x
+    }
+
+    /// Replace one subtree as a child of its parent with another subtree.
+    fn transplant(&mut self, u: NodeIndex<Ix>, v: NodeIndex<Ix>) {
+        if self.parent_ref(u, Node::is_sentinel) {
+            self.root = v;
+        } else if self.is_left_child(u) {
+            self.parent_mut(u, Node::set_left(v));
+        } else {
+            self.parent_mut(u, Node::set_right(v));
+        }
+        self.node_mut(v, Node::set_parent(self.node_ref(u, Node::parent)));
+    }
+
+    /// Check if a node is a left child of its parent.
+    fn is_left_child(&self, node_idx: NodeIndex<Ix>) -> bool {
+        self.parent_ref(node_idx, Node::left) == node_idx
+    }
+
+    /// Check if a node is a right child of its parent.
+    fn is_right_child(&self, node_idx: NodeIndex<Ix>) -> bool {
+        self.parent_ref(node_idx, Node::right) == node_idx
+    }
+
+    /// Update nodes indices after remove
+    ///
+    /// This method has a time complexity of `O(logn)`, as we need to
+    /// update the max index from bottom to top.
+    fn update_idx(&mut self, old: NodeIndex<Ix>, new: NodeIndex<Ix>) {
+        if self.root == old {
+            self.root = new;
+        }
+        if self.nodes.get(new.index()).is_some() {
+            if !self.parent_ref(new, Node::is_sentinel) {
+                if self.parent_ref(new, Node::left) == old {
+                    self.parent_mut(new, Node::set_left(new));
+                } else {
+                    self.parent_mut(new, Node::set_right(new));
+                }
+            }
+            self.left_mut(new, Node::set_parent(new));
+            self.right_mut(new, Node::set_parent(new));
+
+            let mut p = new;
+            while !self.node_ref(p, Node::is_sentinel) {
+                if self.node_ref(p, Node::max_index) == old {
+                    self.node_mut(p, Node::set_max_index(new));
+                }
+                p = self.node_ref(p, Node::parent);
+            }
+        }
+    }
+}
+
+#[cfg(feature = "graphviz")]
+impl<T, V, Ix> IntervalMap<T, V, Ix>
+where
+    T: Ord + Copy + Display,
+    V: Display,
+    Ix: IndexType,
+{
+    /// writes dot file to `filename`. `T` and `V` should implement `Display`.
+    pub fn draw(&self, filename: &str) -> std::io::Result<()> {
```

Comment:
Is this draw feature mainly used for debugging?

Reply:
It could be. Actually, I add it only because iset has the feature.